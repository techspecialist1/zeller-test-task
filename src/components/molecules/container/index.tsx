import React, {FC} from 'react';
import {SafeAreaView, View} from 'react-native';

import {styles} from './styles';

type ContainerProps = {
  children: React.ReactNode;
};

const Container: FC<ContainerProps> = ({children}) => {
  return (
    <SafeAreaView style={styles.safeArea}>
      <View style={styles.container}>{children}</View>
    </SafeAreaView>
  );
};

export default Container;
