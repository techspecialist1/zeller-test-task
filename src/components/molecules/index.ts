import ListView from './listView';
import Container from './container';
import RadioButton from '../atoms/radioButton';
import SearchInput from '../atoms/searchInput';
import ActivityIndicator from '../atoms/activityIndicator';
import ErrorBox from '../atoms/errorBox';

export {
  ListView,
  ActivityIndicator,
  RadioButton,
  SearchInput,
  Container,
  ErrorBox,
};
