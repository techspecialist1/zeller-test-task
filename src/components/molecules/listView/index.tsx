import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';

import {styles} from './style';

type ListViewProps = {
  item: {
    name: string;
    role: string;
  };
  onPress: () => void;
};

const ListView: React.FC<ListViewProps> = props => {
  const {item, onPress} = props;

  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.userList}>
        <View style={styles.profile}>
          <Text style={styles.profileText}>{item?.name?.slice(0, 1)}</Text>
        </View>
        <View style={styles.listName}>
          <Text style={styles.name}>{item?.name}</Text>
          <Text style={styles.role}>{item?.role}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default ListView;
