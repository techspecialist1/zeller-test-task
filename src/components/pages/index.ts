import HomePage from './homePage';
import DetailPage from './detailPage';

export {HomePage, DetailPage};
