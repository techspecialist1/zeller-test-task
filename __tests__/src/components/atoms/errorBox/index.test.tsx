import React from 'react';
import renderer from 'react-test-renderer';
import {ErrorBox} from '../../../../../src/components/molecules';
import {Text} from 'react-native';

describe('ErrorBox', () => {
  test('renders correctly with given text', () => {
    const text = 'An error occurred';
    const component = renderer.create(<ErrorBox text={text} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('renders the correct text content', () => {
    const text = 'Another error message';
    const component = renderer.create(<ErrorBox text={text} />);
    const instance = component.root;
    const textElement = instance.findByType(Text);

    expect(textElement.props.children).toBe(text);
  });
});
