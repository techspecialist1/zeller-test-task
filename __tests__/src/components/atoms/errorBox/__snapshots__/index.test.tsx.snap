// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox renders correctly with given text 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 12.307692307692307,
    }
  }
>
  <Text
    style={
      {
        "color": "#FF0000",
      }
    }
  >
    An error occurred
  </Text>
</View>
`;
